### PR TITLE
test: Test ReadOnlyHistoryAdmin

### LINKED ISSUE #1604: Automated tests for missing coverage in tcms.core.history.ReadOnlyHistoryAdmin
Kiwi TCMS needs additional automated test(s) as mentioned in subject.

**Definition of Done**:
- full coverage for the module or class mentioned in the title. This means both positive/negative scenarios where this makes sense or scenarios which traverse different branches of the code where that makes sense.
- test class/test method names are descriptive and reflect the behavior of the subject under test.  A good scheme to follow is `Test<ClassUnderTest>.test_[who][what][expected_results]`
- CI pass
- All code review passed & PR merged

Hints:
- Coverage report is at https://codecov.io/gh/kiwitcms/Kiwi/tree/master/tcms
- Devel getting started docs: https://kiwitcms.readthedocs.io/en/latest/contribution.html#testing
- Django testing documentation: https://docs.djangoproject.com/en/3.0/topics/testing/

*This issue is part of Kiwi TCMS open source bounty program. For more information see the link(s) in [bounty-program milestone](https://github.com/kiwitcms/Kiwi/milestone/4)*

### PR DESCRIPTION
**Pull request by devkapilbansal**
Fixes #1604

### CODE REVIEW

File: tcms/testplans/tests/test_admin.py [hunk 1]
```diff
@@ -40,3 +40,37 @@ def test_users_can_delete_testplan(self):
...
+            )
+
+        with self.subTest("Test Reverting Redirect to Object"):
+            response = self.client.post(
+                "/admin/testplans/testplan/%d/history/1/" % self.test_plan.pk,
+                follow=True,
+            )
+            self.assertRedirects(
+                response,
+                reverse("test_plan_url", args=[self.test_plan.pk, old_test_plan_name]),
```
Comment (atodorov) [inline tcms/testplans/tests/test_admin.py:75]: The second argument is not the name verbatim but a slug. The URL regex imposes limitation to the available characters that's why the assertion fails.

See also `self.test_plan._get_absolute_url()`

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/testplans/tests/test_admin.py [hunk 2]
```diff
@@ -40,3 +40,34 @@ def test_users_can_delete_testplan(self):
...
+            )
+
+        with self.subTest("Test Reverting Redirect to Object"):
+            response = self.client.post(
+                "/admin/testplans/testplan/%d/history/1/" % self.test_plan.pk,
+                follow=True,
+            )
+            self.assertRedirects(
+                response,
+                self.test_plan._get_absolute_url(),
```
Comment (atodorov) [inline tcms/testplans/tests/test_admin.py:72]: This method is dynamic, it evaluates the URL based on the latest name (e.g. after the save).  Grab the URL before you make any changes and reversals.

Comment (atodorov) [reply tcms/testplans/tests/test_admin.py:72]: `_get_absolute_url()` -> `get_absolute_url()` to make pylint happy.

**Review by atodorov — CHANGES_REQUESTED** [review]

**Comment by devkapilbansal** [comment]
When testing in different language, why the response code is 404?
```
======================================================================
FAIL: test_history_display_testplan (tcms.testplans.tests.test_admin.TestTestPlanAdmin) [Test fields are read-only]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/Kiwi/Kiwi/tcms/testplans/tests/test_admin.py", line 72, in test_history_display_testplan
    '<input type="text"',
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/django/test/testcases.py", line 479, in assertNotContains
    response, text, status_code, msg_prefix, html)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/django/test/testcases.py", line 434, in _assert_contains
    " (expected %d)" % (response.status_code, status_code)
AssertionError: 404 != 200 : Couldn't retrieve content: Response code was 404 (expected 200)

======================================================================
FAIL: test_history_display_testplan (tcms.testplans.tests.test_admin.TestTestPlanAdmin) [Test Reverting Redirect to Object]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/Kiwi/Kiwi/tcms/testplans/tests/test_admin.py", line 82, in test_history_display_testplan
    old_test_plan_url,
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/django/test/testcases.py", line 336, in assertRedirects
    % (response.status_code, status_code)
AssertionError: [] is not true : Response didn't redirect as expected: Response code was 404 (expected 302)
```

**Comment by atodorov** [comment]
> When testing in different language, why the response code is 404?
> 

404 is page not found. As to why you need to debug and figure it out. Print the content returned in the response and also experiment with and without the `follow=True` argument to see what happens.

**Comment by devkapilbansal** [comment]
@atodorov this PR is ready to review now.

**Pull request merged**